context/index.js

```diff
@@ -1,38 +1,59 @@
-import React, { createContext, useState, useEffect } from 'react'
-import Axios from 'axios'
+import React, { createContext, useState, useEffect } from 'react';
+import AsyncStorage from '@react-native-async-storage/async-storage';
+import Axios from 'axios';
 
-export const AppContext = createContext()
+export const AppContext = createContext();
 
 export const AppContextProvider = (props) => {
-  console.log('hello from appContext')
+  console.log('hello from appContext');
 
-  const [categoryData, setCategoryData] = useState([])
-  const [triviaData, setTriviaData] = useState([])
-  const [id, setId] = useState()
-  const [difficulty, setDifficulty] = useState()
+  const [categoryData, setCategoryData] = useState([]);
+  const [triviaData, setTriviaData] = useState([]);
+  const [id, setId] = useState();
+  const [difficulty, setDifficulty] = useState();
 
-  // useEffect(() => {
-  //   const getAllCategoryData = async() => {
-  //     Axios.get('https://opentdb.com/api_category.php').then((response) => {
-  //       console.log("respons", response)
-  //       // setCategoryData(response.data.trivia_categories)
-  //     })
-  //   }
-  //   getAllCategoryData()
-  // }, [])
+  const getAllCategoryData = async () => {
+    const categories = await AsyncStorage.getItem('categories');
```

Comment:
Nice, this will reduce the number of API calls for the category as you are storing in local storage. I'm good with this but wouldn't this logic down the road not allow for new categories to come in if they are added in, assuming the user does not reset their cache? Maybe a workaround can be to compare the length of categories on the first launch of the app and if the same return?